Cargo.toml

```diff
@@ -11,6 +11,8 @@ license = "MIT OR Apache-2.0"
 repository = "https://github.com/dtolnay/thiserror"
 rust-version = "1.56"
 
+# without the `std` feature (i.e. `no_std`): rust-version = "1.81"
```

Comment:
As mentioned in [this comment](https://github.com/dtolnay/thiserror/pull/304#issuecomment-2407101161), I feel we should attempt to widen our build compatibility to not break `rustc < 1.81` builds importing `thiserror` with `default-features = false`.